app/controllers/api/v1/trends/links_controller.rb

```diff
@@ -1,13 +1,16 @@
 # frozen_string_literal: true
 
 class Api::V1::Trends::LinksController < Api::BaseController
+  vary_by 'Authorization, Accept-Language'
+
   before_action :set_links
 
   after_action :insert_pagination_headers
 
   DEFAULT_LINKS_LIMIT = 10
 
   def index
+    cache_if_unauthenticated!
```

Comment:
This is very likely to cause inconsistent page cache, especially since trends pagination does not work with an underlying ID and depends on the rank of the trend items.